docs/how-to/how-to-use-template-in-runtime-rs.md

```diff
@@ -0,0 +1,108 @@
+# How to Use Template in runtime-rs
+
+## What is VM Templating
+
+VM templating is a Kata Containers feature that enables new VM creation using a cloning technique. When enabled, new VMs are created by cloning from a pre-created template VM, and they will share the same initramfs, kernel and agent memory in readonly mode. It is very much like a process fork done by the kernel but here we *fork* VMs.
+
+For more details on VM templating, refer to the [What is VM templating and how do I use it](./what-is-vm-templating-and-how-do-I-use-it.md) article.
+
+## How to Enable VM Templating
+
+VM templating can be enabled by changing your Kata Containers config file (`/usr/share/defaults/kata-containers/configuration.toml`, overridden by `/etc/kata-containers/configuration.toml` if provided) such that:
+
+- `qemu` version `v4.1.0` or above is specified in `hypervisor.qemu`->`path` section
+- `enable_template = true`
+- `template_path = "/run/vc/vm/template"`
+- `initrd =` is set
+- `image =` option is commented out or removed
+- `shared_fs =` option is commented out or removed
+- `default_memory =` should be set to more than 256MB
+
+Then you can create a VM template for later usage by calling:
+```bash
+sudo kata-ctl factory init
+```
+
+Check status by calling:
+
+```bash
+sudo kata-ctl factory status
+```
+
+Purge it by calling:
+
+```bash
+sudo kata-ctl factory destroy
+```
+
+## How to Create a New VM from VM Template
+
+Due to a conflict between the VM template mechanism and VirtioFS in memory sharing, we have opted to use `snapshotter` instead of `shared-fs` to provide the rootfs for containers. Therefore, we need to modify the containerd configuration to use snapshotter.
```

Comment:
You'd better explain the difference between runtime-go and runtime-rs on the point why we use snapshotter(specially blockfile snapshotter), and why not use 9pfs(9pfs not supported in runtime-rs) and why we don't use shared-fs ?